crates/router/src/connector/bankofamerica/transformers.rs

```diff
@@ -63,105 +189,385 @@ impl TryFrom<&BankofamericaRouterData<&types::PaymentsAuthorizeRouterData>>
-    Processing,
+    Voided,
+    Reversed,
+    Pending,
+    Declined,
+    AuthorizedPendingReview,
+    Transmitted,
 }
 
-impl From<BankofamericaPaymentStatus> for enums::AttemptStatus {
-    fn from(item: BankofamericaPaymentStatus) -> Self {
-        match item {
-            BankofamericaPaymentStatus::Succeeded => Self::Charged,
-            BankofamericaPaymentStatus::Failed => Self::Failure,
-            BankofamericaPaymentStatus::Processing => Self::Authorizing,
+impl ForeignFrom<(BankofamericaPaymentStatus, bool)> for enums::AttemptStatus {
+    fn foreign_from(item: (BankofamericaPaymentStatus, bool)) -> Self {
+        let status = match item.0 {
+            BankofamericaPaymentStatus::Authorized
+            | BankofamericaPaymentStatus::AuthorizedPendingReview => Self::Authorized,
+            BankofamericaPaymentStatus::Succeeded | BankofamericaPaymentStatus::Transmitted => {
+                Self::Charged
+            }
+            BankofamericaPaymentStatus::Voided | BankofamericaPaymentStatus::Reversed => {
+                Self::Voided
+            }
+            BankofamericaPaymentStatus::Failed | BankofamericaPaymentStatus::Declined => {
+                Self::Failure
+            }
+            BankofamericaPaymentStatus::Pending => Self::Pending,
+        };
+        if item.1 && status == Self::Authorized {
+            return enums::AttemptStatus::Pending;
         }
+        status
     }
 }
 
-//TODO: Fill the struct with respective fields
-#[derive(Default, Debug, Clone, Serialize, Deserialize, PartialEq)]
+#[derive(Debug, Deserialize)]
 pub struct BankofamericaPaymentsResponse {
-    status: BankofamericaPaymentStatus,
     id: String,
+    status: BankofamericaPaymentStatus,
+    error_information: Option<BankofamericaErrorInformation>,
+    client_reference_information: Option<ClientReferenceInformation>,
+}
+
+#[derive(Debug, Deserialize)]
+pub struct BankofamericaErrorInformation {
+    reason: String,
+    message: String,
 }
 
-impl<F, T>
+impl<F>
     TryFrom<
-        types::ResponseRouterData<F, BankofamericaPaymentsResponse, T, types::PaymentsResponseData>,
-    > for types::RouterData<F, T, types::PaymentsResponseData>
+        types::ResponseRouterData<
+            F,
+            BankofamericaPaymentsResponse,
+            types::PaymentsAuthorizeData,
+            types::PaymentsResponseData,
+        >,
+    > for types::RouterData<F, types::PaymentsAuthorizeData, types::PaymentsResponseData>
 {
     type Error = error_stack::Report<errors::ConnectorError>;
     fn try_from(
         item: types::ResponseRouterData<
             F,
             BankofamericaPaymentsResponse,
-            T,
+            types::PaymentsAuthorizeData,
+            types::PaymentsResponseData,
+        >,
+    ) -> Result<Self, Self::Error> {
+        Ok(Self {
+            status: enums::AttemptStatus::foreign_from((
+                item.response.status,
+                item.data.request.is_auto_capture()?,
```

Comment:
This is not the right approach to decide status. We can discuss on this, after we get clarity from connector about the status

Reply:
Done